drivers: display: support display driver on Renesas RA EK-RA8D1

thenguyenyf opened the pull request:
- Add support for MIPI-DSI driver
- Add support for Display driver
- Add support for rtkmipilcdb00000be display shield
- Add support for SDRAM on EK-RA8D1

danieldegrasse commented on soc/renesas/ra/ra8d1/sections.ld:
```diff
@@ -77,3 +77,11 @@ SECTION_PROLOGUE(.option_setting_s,,)
+{
+	__SDRAM_Start = .;
+	KEEP(*(.sdram*))
+    KEEP(*(.lvgl_buf*))
```
Not sure this should be at the SOC level. Will every developer using this SOC want to link LVGL buffers into SDRAM? Perhaps they are driving a small display, and their board has no SDRAM. I think this should be at the board level, like we do for the MIMXRT595 for example: https://github.com/zephyrproject-rtos/zephyr/blob/5dc5fa5ee2ac4c0b3914ce926d1e126f052d1e99/boards/nxp/mimxrt595_evk/dc_ram.ld#L14

thenguyenyf replied:
Done. I moved it into board layer

danieldegrasse commented on dts/arm/renesas/ra/ra8/r7fa8d1bhecbd.dtsi:
```diff
@@ -25,5 +25,12 @@
 			};
 		};
+
+		sdram: sdram@68000000 {
```
Again here- should this be at the SOC level? Seems like the SDRAM on the EK-RA8D1 is an external memory chip specific to that board. Perhaps the reg address can be set here, but the size probably needs to be 0, and can be overridden at the board level.

thenguyenyf replied:
Done

danieldegrasse replied:
I think the best option here is probably one of these two approaches:

- Define the *entire* SDRAM node at the board level
- Define a node for the SDRAM controller on the SOC at the SOC level, and define the SDRAM part itself at the board level. This is similar to how we define other external devices, like flash chips on a FlexSPI bus.

thenguyenyf replied:
Done. I have updated as first one. Update for SDRAM controller as second one will be added later. Thanks for your idea.

danieldegrasse commented on boards/shields/rtkmipilcdb00000be/Kconfig.defconfig:
```diff
@@ -0,0 +1,54 @@
+
+if LVGL
+
+# Configure LVGL to use touchscreen with KSCAN API
```
This comment isn't correct, we are using the input API here.

danieldegrasse commented on boards/shields/rtkmipilcdb00000be/Kconfig.defconfig:
```diff
@@ -0,0 +1,54 @@
+if SHIELD_RTKMIPILCDB00000BE
+if DISPLAY
+
+config INPUT_INIT_PRIORITY
```
Please add a comment as to why the input priority needs to be changed

thenguyenyf replied:
Done

danieldegrasse commented on boards/shields/rtkmipilcdb00000be/Kconfig.defconfig:
```diff
@@ -0,0 +1,54 @@
+config LV_Z_DOUBLE_VDB
+	default y
+
+# Force full refresh. This prevents memory copy associated with partial
```
This comment is probably coming from this file, right?

https://github.com/zephyrproject-rtos/zephyr/blob/5dc5fa5ee2ac4c0b3914ce926d1e126f052d1e99/boards/shields/rk055hdmipi4m/Kconfig.defconfig#L31

If you need this to be enabled, update the comment to refer to the LCD controller on the Renesas board.

danieldegrasse commented on boards/shields/rtkmipilcdb00000be/Kconfig.defconfig:
```diff
@@ -0,0 +1,54 @@
+config INPUT
+	default y
+
+config INPUT_GT911_INTERRUPT
```
Not needed. This Kconfig already defaults to disabled

thenguyenyf replied:
I fixed it

danieldegrasse commented on boards/shields/rtkmipilcdb00000be/doc/index.rst:
```diff
@@ -0,0 +1,61 @@
+This display uses a 26 pin connector header.
+
+Pins Assignment of the Renesas RTKMIPILCDB00000BE MIPI Display
+===============================================================
```
Underlines need to be the same width as the title (it looks like you have an extra `=` here, at least in the github UI)

danieldegrasse commented on boards/shields/rtkmipilcdb00000be/doc/index.rst:
```diff
@@ -0,0 +1,61 @@
+.. _rtkmipilcdb00000be:
+
+RTKMIPILCDB00000BE MIPI Display
+############################
```
Underlines need to be same width as the title. This is likely why your documentation build is failing.

danieldegrasse commented on boards/shields/rtkmipilcdb00000be/doc/index.rst:
```diff
@@ -0,0 +1,61 @@
+Programming
+***********
+
+Set ``-DSHIELD=rtkmipilcdb00000be`` when you invoke ``west build``. For
```
```suggestion
Set ``--shield=rtkmipilcdb00000be`` when you invoke ``west build``. For
```

thenguyenyf replied:
I fixed it

danieldegrasse commented on drivers/display/display_ili9806e_dsi.c:
```diff
@@ -0,0 +1,440 @@
+
+	/* Loop trough all commands as long as writes are successful*/
+	for (int i = 0; i < nr_cmds && ret == 0; i++) {
+		ret = ili9806e_write_reg(dev, cmd->reg, cmd->cmd, cmd->cmd_len);
```
Check `ret` here, and return if it is nonzero.

thenguyenyf replied:
I fixed it

danieldegrasse commented on drivers/display/display_ili9806e_dsi.c:
```diff
@@ -0,0 +1,440 @@
+	return ili9806e_write_reg(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+}
+
+static int ili9806e_write(const struct device *dev, uint16_t x, uint16_t y,
```
Why implement this function? If it is absent, the display driver API will still return an error- is this present simply for the error log?

thenguyenyf replied:
Done. I removed this function

danieldegrasse commented on drivers/display/display_ili9806e_dsi.c:
```diff
@@ -0,0 +1,440 @@
+	struct mipi_dsi_device mdev;
+	int ret;
+
+#if !CONFIG_INPUT_GT911
```
This *cannot* be present here. Not every display module with a ili9806e_dsi controller will have a GT911 touch controller.

This looks like a workaround specific to the display module you are using. Could you go into more detail as to why this exists so we can figure out where to put this code? It looks like the ILI9806E and GT911 module might share a reset line? In that case is is possible to set init priorities so only the first device to initialize actually pulses the reset line?

thenguyenyf replied:
> It looks like the ILI9806E and GT911 module might share a reset line

It's correct. In this shield, first the gt911 has to be reset, then the ili9806e has been reset before and skips this reset handler code

danieldegrasse replied:
It is correct for the *shield*, not for the display controller. The issue here is that someone could build a module with a FT5336 touchscreen controller, and ILI9806E display controller- the GT911 and ILI9806E sharing a reset line is a decision specific to the *shield/module*.

If you need to have the GT911 driver pulse the reset line, then have the ILI9806E skip the reset pulse, simply do not define a reset GPIO for the ILI9806E devicetree node, and make sure the GT911 driver starts first, so it can pulse the reset line.

danieldegrasse commented on drivers/display/display_ili9806e_dsi.c:
```diff
@@ -0,0 +1,440 @@
+		.pixel_format = DT_INST_PROP(n, pixel_format),                                     \
+	};                                                                                         \
+	DEVICE_DT_INST_DEFINE(n, &ili9806e_init, NULL, NULL, &ili9806e_config_##n, POST_KERNEL,    \
+			      CONFIG_APPLICATION_INIT_PRIORITY, &ili9806e_api);
```
This is yet another instance of a DSI display driver, where we need the init priority to be lower than the display and MIPI DSI driver.

Perhaps we need to define a second display init priority for parallel mode LCD controllers? It seems like multiple vendors now are using the following display chain:

Parallel LCD controller -> Parallel LCD to DSI video mode adapter -> Display controller

The init priorities need to be like so:
- Parallel LCD controller
- DSI video mode adapter
- DSI display controller driver

I think we should create a custom init priority for the parallel LCD controller drivers to use. These drivers need to start before the DSI video mode adapter, and by default both drivers share an init priority, so either could start first.

danieldegrasse commented on drivers/display/display_ili9806e_dsi.h:
```diff
@@ -0,0 +1,23 @@
+#define ZEPHYR_DRIVERS_DISPLAY_ILITEK_ILI9806E_H_
+
+/* Horizontal low pulse width */
+#define ILITEK_ILI9806E_HSYNC 2U
```
Are these hardcoded? Seems like it would vary based on the panel the controller was driving, and like these should be in devicetree.

thenguyenyf replied:
Done. This configuration has been moved to dts

danieldegrasse commented on drivers/display/display_renesas_ra.c:
```diff
@@ -0,0 +1,317 @@
+	l_hstride = data->display_fsp_cfg.input[0].hstride;
+	buffer_write = (pixel_t *)data->display_fsp_cfg.input[0].p_base;
+
+	for (uint32_t row = 0; row < desc->height; row++) {
```
Ok, a few things here. Please feel free to ignore this comment if I'm misunderstanding the hardware, these are just suggestions :) 

This looks like a display that uses a memory mapped buffer, and streams it to the display. If so, I'd suggest the following:
- Add code to detect if the x, y, and `desc` parameters describe a buffer the same size as the display buffer. If so, pass that buffer directly to hardware. The Kconfigs you have setup for LVGL mean that you will get buffers the size of the screen passed to this function, and using the buffer directly should improve performance
- Add a Kconfig of some kind to allow users to disable allocation of the `fb_background##id` buffer. If a user knows the hardware they are working with, they will probably want to just pass full sized framebuffers to the display, and never use the driver-allocated framebuffer, making this wasted memory.

thenguyenyf replied:
> Add code to detect if the x, y, and desc parameters describe a buffer the same size as the display buffer. If so, pass that buffer directly to hardware. The Kconfigs you have setup for LVGL mean that you will get buffers the size of the screen passed to this function, and using the buffer directly should improve performance

Done

> Add a Kconfig of some kind to allow users to disable allocation of the fb_background##id buffer. If a user knows the hardware they are working with, they will probably want to just pass full sized framebuffers to the display, and never use the driver-allocated framebuffer, making this wasted memory.

Done. I added RENESAS_RA_GLCDC_FB_NUM to handle driver buffer allocation

danieldegrasse commented on dts/arm/renesas/ra/ra8/r7fa8d1bhecbd.dtsi:
```diff
@@ -32,5 +33,23 @@
+			interrupts = <72 1>, <73 1>, <74 1>, <75 1>, <76 1>, <77 1>;
+			interrupt-names = "sq0", "sq1", "vm", "rcv", "ferr", "ppi";
+			clocks = <&lcdclk MSTPC 10>;
+			status = "okay";
```
Don't set nodes to `status = "okay"` at SOC level, keep them disabled here and enable at the board level

danieldegrasse commented on drivers/mipi_dsi/Kconfig.renesas_ra:
```diff
@@ -0,0 +1,13 @@
+	bool "Renesas RA MIPI-DSI Host Controller"
+	default y
+	depends on DT_HAS_RENESAS_RA_MIPI_DSI_ENABLED
+	depends on DT_HAS_RENESAS_RA_DISPLAY_ENABLED
```
Is this second dependency accurate? By this I mean can this MIPI DSI driver support command mode? If so, you probably *don't* need the parallel LCD controller driver to be enabled for all cases where you'd use the DSI driver.

thenguyenyf replied:
Done. I removed this depend on condition

danieldegrasse commented on drivers/mipi_dsi/dsi_renesas_ra.c:
```diff
@@ -0,0 +1,338 @@
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/sys/printk.h>
```
Why is `printk` needed?

thenguyenyf replied:
I removed this inclussion

danieldegrasse commented on modules/lvgl/lvgl.c:
```diff
@@ -152,7 +152,11 @@ static int lvgl_allocate_rendering_buffers(lv_disp_drv_t *disp_driver)
 		buf_size = 4 * buf_nbr_pixels;
 		break;
 	case PIXEL_FORMAT_RGB_888:
+#if defined(CONFIG_RENESAS_RA_DISPLAY)
```
Don't add driver specific Kconfigs here like this. Why does the display hardware need an additional byte per pixel? Is it using RGB_888 as the display format, or ARGB_8888?

thenguyenyf replied:
The RGB888 input buffer for the Renesas GLCD controller driver requires 32 bits as 1 pixel as shown in the image below. So, each RGB888 pixel buffer should have 1 byte padding at MSB.
![image](https://github.com/user-attachments/assets/010b3900-562d-4e70-a555-00ee7d0488ec)
Because handling the memory padding by software inside the driver code can take a huge CPU load, I think it is better to give it to application prepare/input directly 32bitRGB888 buffer for display API. If it is not acceptable, could you suggest a more suitable way?

danieldegrasse replied:
Could the display format be reported as ARGB8888? It seems like the GLCD would just ignore the upper 8 bits, so and ARGB8888 buffer would be equivalent to an RGB8888 buffer with no alpha channel. It seems like LVGL already simply sets the alpha channel to 0xFF in some cases: https://github.com/zephyrproject-rtos/lvgl/blob/zephyr/src/draw/lv_img_buf.c#L51

Given this, why not just report ARGB8888?

thenguyenyf replied:
Done. I moved the processing into the shim driver code. I have confirmed that the LVGL render buffer fills channel A with 0xFF

danieldegrasse commented on modules/lvgl/lvgl_display.c:
```diff
@@ -85,9 +85,15 @@ int set_lvgl_rendering_cb(lv_disp_drv_t *disp_drv)
 #endif
 		break;
 	case PIXEL_FORMAT_RGB_888:
+#if defined(CONFIG_RENESAS_RA_DISPLAY)
```
Same story here. It seems like the display is using ARGB8888?

thenguyenyf replied:
This change has been reverted

danieldegrasse commented on samples/shields/rtkmipilcdb00000be_ra_display/README.rst:
Why does this sample need to exist? What can it do that the display sample we currently have doesn't?

thenguyenyf replied:
There is no special operation, just a sample using 32bitRGB888 buffer with Renesas RA GLCDC

danieldegrasse replied:
Ok- I'd prefer we standardize on using the current display sample then, and not add this. IMO we should not have multiple samples in Zephyr that effectively do the same thing- if we do, it isn't as clear to users (or developers) which sample should be used for testing and enabling new devices

thenguyenyf replied:
I removed this sample and added support to build for samples/drivers/display

danieldegrasse commented on west.yml:
Commit updating Renesas HAL should be first in series to make sure everything is bisectable

thenguyenyf replied:
I fixed it

danieldegrasse reviewed (dismissed):
Commits need to be split up here to be more granular (this suggested list is ordered from oldest to newest)
- Hal update commit
- Commit adding SDRAM memc driver
- Commit adding SDRAM enablement changes at SOC level
- Commit enabling SDRAM driver on EK-RA8D1 board
- Commit enabling memc test for EK-RA8D1
- Commit adding renesas_ra display driver (and dts bindings)
- Commit adding SOC level definitions to support renesas_ra display driver
- Commit adding dsi_renesas_ra driver (and dts bindings)
- Commit adding SOC level definitions to support dsi_renesas_ra driver
- Commit adding ili9806e_dsi display driver (and dts bindings)
- Commit adding board support for renesas_ra, dsi_renesas, and ili9806_dsi display driver
- Commit adding rtkmipilcdb00000be shield
- Commit adding display test support for the EK-RA8D1

Note- the change request here is specifically about the use of `CONFIG_RENESAS_RA_DISPLAY` in generic code such as LVGL.

thenguyenyf commented on drivers/memc/memc_renesas_ra_sdram.c:
```diff
@@ -0,0 +1,41 @@
+		LOG_ERR("pin function initial failed");
+		return err;
+	}
+	sdram_init();
```
Current implementation just supports for SDRAM on EK-RA8D1

danieldegrasse replied:
Ok- I would suggest naming the driver based on the SDRAM part number then. Otherwise, users may incorrectly assume that the driver will work for any SDRAM. Another (better) option would be to implement a MEMC driver capable of configuring the SDRAM controller based on settings in devicetree, so it can work for any SDRAM chip

danieldegrasse replied:
I'm still concerned about the naming of this driver. This implies the SDRAM memc driver will enable SDRAM support on any board with an RA SOC, right?

What will happen when a customer designs a board with a different SDRAM part, and tries to use this driver?

One option to explore would be making the SDRAM init part of a board specific C init file.

thenguyenyf committed:
manifest: update rev of hal_renesas to latest

Update hal_renesas rev to add support for these features:
- SDRAM support on EK-RA8D1
- LCD controller
- MIPI DSI

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf committed:
drivers: memc: enable support for SDRAM controller on Renesas RA family

First commit to add support for SDRAM controller on Renesas RA SoC

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf committed:
drivers: display: add support for GLCDC on Renesas RA family

First commit to add support for Graphics LCD Controller on Renesas RA

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf committed:
dts: arm: renesas: add support for GLCDC driver on RA8D1

Add device node to support Graphics LCD Controller on RA8D1 SoC

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf committed:
drivers: mipi_dsi: add support for MIPI DSI driver on Renesas RA family

First commit to add support for MIPI DSI driver on Renesas RA

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf committed:
dts: arm: renesas: add support for MIPI DSI on RA8D1

Add device node to support MIPI DSI driver on Renesas RA8D1 SoC

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf committed:
drivers: display: add support for ili9806e_dsi driver

First commit to add support for ili9806e_dsi driver

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf committed:
boards: shields: add support for rtkmipilcdb00000be shield

First commit to support rtkmipilcdb00000be display shield

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf committed:
samples: display: add support for display sample on ek_ra8d1

Add support to build samples/drivers/display on ek_ra8d1

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf committed:
tests: display: add support test GLCDC on ek_ra8d1

This commit to add support to build tests/drivers/display on ek_ra8d1

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf committed:
samples: lvgl: add support for lvgl demo on ek_ra8d1

This commit to add support to build samples/subsys/display on ek_ra8d1

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf committed:
samples: lvgl: add support for rtkmipilcdb00000be to build lvgl demo

Add support for lvgl demo build with rtkmipilcdb00000be shield

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf commented:
Last push to rebase main branch before running with GitHub CI

danieldegrasse commented on drivers/display/display_ili9806e_dsi.h:
```diff
@@ -0,0 +1,32 @@
+
+#define GLCD_TIMINGS_NODE DT_CHILD(DT_NODELABEL(zephyr_lcdif), display_timings)
+
+#define GLCD_HTIMMING_SYNC       DT_PROP(GLCD_TIMINGS_NODE, hsync_len)
```
Why name these GLCD_TIMINGS? They are not specific to the GLCD.

thenguyenyf replied:
I fixed it

danieldegrasse commented:
@thenguyenyf please make sure you are rebasing on the latest main- also, your documentation build will continue to fail until you fix the underline issues I highlighted in my comments (a few also exist in that RST file)

thenguyenyf commented on samples/modules/lvgl/demos/sample.yaml:
```diff
@@ -58,3 +58,20 @@ tests:
+    extra_args: SHIELD=rtkmipilcdb00000be
+    extra_configs:
+      - CONFIG_MEMC=y
+      - CONFIG_LV_Z_VBD_CUSTOM_SECTION=y
```
Done. I moved it into boards/shields/rtkmipilcdb00000be/Kconfig.defconfig

thenguyenyf commented on samples/subsys/display/lvgl/sample.yaml:
```diff
@@ -92,3 +92,16 @@ tests:
+    extra_args: SHIELD=rtkmipilcdb00000be
+    extra_configs:
+      - CONFIG_MEMC=y
+      - CONFIG_LV_Z_VBD_CUSTOM_SECTION=y
```
Done

thenguyenyf commented on drivers/display/display_renesas_ra.c:
```diff
@@ -0,0 +1,417 @@
+	if (pixel_format == config->pixel_format) {
+		return 0;
+	}
+	LOG_WRN("Pixel format changes must be set in dts at build time.");
```
I think this is not an issue. This warning is for the display controller and with display_ili9806e_dsi for the shield driver. So, it can be understood that:

- This driver should use build-time config on dts to synchronous with the display_ili9806e_dsi config. 
- And display_ili9806e_dsi pixel_format cannot supported (because it uses output format of display controller)

danieldegrasse replied:
This should be a LOG_ERR, not warning IMO

thenguyenyf replied:
Done. I updated it to LOG_ERR.

thenguyenyf commented on drivers/display/Kconfig.renesas_ra:
```diff
@@ -0,0 +1,32 @@
+	default 64
+	help
+	  Rendering buffer alignment.
+
```
Done. This config was removed

danieldegrasse commented on tests/drivers/memc/ram/src/main.c:
```diff
@@ -48,13 +48,26 @@ BUF_DEF(sram2);
 BUF_DEF(psram);
 #endif
 
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(sdram), okay)
```
Does this run on the hardware with twister currently? Looking at the testcase.yaml: https://github.com/zephyrproject-rtos/zephyr/blob/305ceba9204d9a7ee4bae0a6103ef4cbd9a1c54e/tests/drivers/memc/ram/testcase.yaml#L2

It looks like you'd need to add a new entry to get this to run on the EK_RA8D1 (I'd suggest doing it by the DT compatible, like other testcases have done with their filter)

thenguyenyf replied:
Sorry I miss this comment. Updated

danieldegrasse commented on dts/bindings/display/renesas,ra-display.yaml:
```diff
@@ -0,0 +1,57 @@
+    description: |
+      Initial output Pixel format for Graphic LCD controller.
+
+  htiming-total-cyc:
```
Can these be calculated from the display timings node (perhaps using the clock rate of the GLCD), or are they different values?

thenguyenyf replied:
Done. This config has been moved into shim driver code.

danieldegrasse commented on drivers/mipi_dsi/dsi_renesas_ra.c:
```diff
@@ -0,0 +1,337 @@
+	mipi_dsi_cfg_t mipi_dsi_cfg;
+};
+
+volatile bool g_message_sent;
```
Don't use global variables like these. Store these within the device data structure to properly support multi instance drivers

danieldegrasse commented on drivers/display/display_ili9806e_dsi.c:
```diff
@@ -0,0 +1,434 @@
+{
+	int ret = 0;
+
+	/* Loop trough all commands as long as writes are successful*/
```
Nit- spelling

thenguyenyf replied:
Done

uLipe commented on drivers/display/display_ili9806e_dsi.c:
```diff
@@ -0,0 +1,434 @@
+		return 0;
+	}
+	LOG_WRN("Pixel format change not implemented");
+
```
I think this Warning can be called even when the only supported color is passed since this function is noop always.

thenguyenyf replied:
Done

uLipe commented on drivers/display/display_ili9806e_dsi.c:
```diff
@@ -0,0 +1,434 @@
+static const struct display_driver_api ili9806e_api = {
+	.blanking_on = ili9806e_blanking_on,
+	.blanking_off = ili9806e_blanking_off,
+	.set_pixel_format = ili9806e_pixel_format,
```
Since set pixel format api is noop, instead of a warning, it can be removed entirely, the device api if find this function is not implemented it will return -ENOSYS

thenguyenyf replied:
It's referred from some existed driver for display shield (such as [display_hx8394.c](https://github.com/renesas/zephyr/blob/58779db764192c38ba7c52f879f9f9d9f1703d08/drivers/display/display_hx8394.c#L479) or [display_rm68200.c](https://github.com/renesas/zephyr/blob/58779db764192c38ba7c52f879f9f9d9f1703d08/drivers/display/display_rm68200.c#L134)). So, I will consider what is better in that case.

uLipe reviewed (commented):
I have some initial comments

thenguyenyf committed:
dts: arm: renesas: add support for sdram controller on RA8D1

Add device node to support SDRAM controller on Renesas RA8D1 SoC

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf committed:
boards: renesas: enable support for SDRAM on ek_ra8d1 board

Add support for SDRAM on Renesas RA EK-RA8D1

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf commented:
Last push to rebase main and solve conflict

thenguyenyf commented:
Hi @danieldegrasse , @thaoluonguw , @uLipe . Sorry for my delay. Could you please check for updates?

pdgendt commented on drivers/display/display_ili9806e_dsi.c:
```diff
@@ -0,0 +1,434 @@
+	return -ENOTSUP;
+}
+
+static const struct display_driver_api ili9806e_api = {
```
Zephyr has introduced iterable sections for device APIs, and drivers should use the wrapper macros now.

```suggestion
static DEVICE_API(display, ili9806e_api) = {
```

thenguyenyf replied:
I fixed it

pdgendt commented on drivers/display/display_renesas_ra.c:
```diff
@@ -0,0 +1,431 @@
+	return -ENOTSUP;
+}
+
+static const struct display_driver_api display_api = {
```
```suggestion
static DEVICE_API(display, display_api) = {
```

thenguyenyf replied:
I fixed it

pdgendt commented on drivers/mipi_dsi/dsi_renesas_ra.c:
```diff
@@ -0,0 +1,342 @@
+	return len;
+}
+
+static struct mipi_dsi_driver_api mipi_dsi_api = {
```
```suggestion
static DEVICE_API(mipi_dsi, mipi_dsi_api) = {
```

thenguyenyf replied:
I fixed it

danieldegrasse commented on tests/drivers/display/display_read_write/src/main.c:
```diff
@@ -14,8 +14,14 @@ LOG_MODULE_DECLARE(display_api, CONFIG_DISPLAY_LOG_LEVEL);
 static const struct device *dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_display));
 static const uint32_t display_width = DT_PROP(DT_CHOSEN(zephyr_display), width);
 static const uint32_t display_height = DT_PROP(DT_CHOSEN(zephyr_display), height);
+#if defined(CONFIG_RENESAS_RA_GLCDC)
```
Instead of making this specific to the GLCDC, why not add KConfigs for this sample to set the section name and alignment of the display buffer? Then these can be setup in a board overlay for the Renesas boards

thenguyenyf replied:
I fixed it

danieldegrasse commented on tests/drivers/display/display_read_write/src/main.c:
```diff
@@ -26,7 +32,11 @@ static inline uint8_t bytes_per_pixel(enum display_pixel_format pixel_format)
 	case PIXEL_FORMAT_ARGB_8888:
 		return 4;
 	case PIXEL_FORMAT_RGB_888:
+#if defined(CONFIG_RENESAS_RA_GLCDC)
```
We cannot do this. Weren't we able to report the `ARGB_8888` pixel format from the driver to resolve this same issue in the LVGL glue code?

thenguyenyf replied:
I revert this change

thenguyenyf commented on drivers/memc/memc_renesas_ra_sdram.c:
```diff
@@ -0,0 +1,41 @@
+	.pincfg = PINCTRL_DT_INST_DEV_CONFIG_GET(0),
+};
+
+DEVICE_DT_INST_DEFINE(0, renesas_ra_sdram_init, NULL, NULL, &config, POST_KERNEL,
```
These properties have been used for configuration macro in HAL layer (hal_renesas module). So in this shim driver we just need to call SDRAM BSP API.

thenguyenyf commented on modules/Kconfig.renesas_fsp:
```diff
@@ -98,4 +98,19 @@ config USE_RA_FSP_ETHER
 	help
 	  Enable RA FSP Ethernet driver
 
+config USB_RA_FSP_SDRAM
```
I corrected this typo. I think this config should be keep as it as, for hal_renesas config consistency, and maybe easier to maintain/extend support for Renesas FSP SDRAM in the future.

thenguyenyf commented:
Last push to rebase main and solve conflict

thenguyenyf committed:
tests: memc: ram: support test for sdram controller on Renesas RA family

Add SDRAM controller test support

Signed-off-by: The Nguyen <the.nguyen.yf@renesas.com>

thenguyenyf commented:
Last push to update manifest revision because it was merged

thenguyenyf commented:
Hi @KhiemNguyenT , @soburi , @thaoluonguw , @duynguyenxa . Could you please help to take a look?

uLipe reviewed (commented):
LGTM!

kartben commented:
This has a merge conflict

thenguyenyf commented:
Hi @danieldegrasse . Could you please take a look?

danieldegrasse commented on tests/drivers/display/display_read_write/Kconfig:
```diff
@@ -0,0 +1,27 @@
+
+config DISPLAY_BUFFER_USE_GENERIC_SECTION
+	bool "Place the display buffer in a specific memory section"
+	default n
```
Nit- redundant `default n`

thenguyenyf replied:
Fixed. Thanks for your support!

danieldegrasse reviewed (commented):
Just a minor nit- dismissing my block as it has been addressed. Otherwise LGTM

uLipe reviewed (approved):
Great job!
